tests/mock_network/mod.rs

```diff
@@ -65,6 +66,55 @@ fn nodes_with_prefix_mut<'a>(
         .filter(move |node| prefix.matches(&node.name()))
 }
 
+pub fn count_sections_members_if_split(nodes: &[TestNode]) -> BTreeMap<Prefix<XorName>, usize> {
+    let mut counts = BTreeMap::new();
+    for pfx in nodes.iter().map(node_prefix_if_split) {
+        *counts.entry(pfx).or_default() += 1;
+        let _ = counts.entry(pfx.sibling()).or_default();
+    }
+    counts
+}
+
+pub fn node_prefix_if_split(node: &TestNode) -> Prefix<XorName> {
+    let prefix = node.our_prefix();
+
+    let sub_prefix = [prefix.pushed(false), prefix.pushed(true)]
+        .iter()
+        .find(|ref pfx| pfx.matches(&node.name()))
+        .cloned();
+    unwrap!(sub_prefix)
+}
+
+fn new_node_prefix_without_split(
+    node: &TestNode,
+    count_if_split_node: &BTreeMap<Prefix<XorName>, usize>,
+    min_split_size: usize,
+) -> Option<Prefix<XorName>> {
+    if let Some((sub_prefix, count)) = count_if_split_node
+        .iter()
+        .find(|(pfx, _)| pfx.matches(&node.name()))
+    {
+        if *count < min_split_size * 2 - 1 {
+            return Some(*sub_prefix);
+        }
+    }
+    None
+}
+
+fn can_accept_node_without_split(
+    count_if_split_node: &BTreeMap<Prefix<XorName>, usize>,
+    min_split_size: usize,
+) -> bool {
+    count_if_split_node
+        .values()
+        .any(|count| *count < min_split_size * 2 - 1)
```

Comment:
So this means `a node can be accepted without split when ANY of the input prefix contains less than 2*min_size-1 nodes` ?
I thought shall be `ALL` ? or we return someting like `Some(prefix)` ?

Reply:
If any are `*count < min_split_size * 2 - 1`, then at least one sub prefix can accept more nodes.
if none do, then there is no node we can create that could be accepted so we avoid the infinite loop.